tests/hw-accel/internal/deploy/operator-installer.go

```diff
@@ -0,0 +1,317 @@
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+	"github.com/openshift-kni/eco-goinfra/pkg/olm"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// DefaultLogLevel for operator installation logging.
+	DefaultLogLevel = 90
+)
+
+// OperatorInstallConfig holds configuration for operator installation.
+type OperatorInstallConfig struct {
+	APIClient              *clients.Settings
+	Namespace              string
+	OperatorGroupName      string
+	SubscriptionName       string
+	PackageName            string
+	CatalogSource          string
+	CatalogSourceNamespace string
+	Channel                string
+	SkipNamespaceCreation  bool
+	SkipOperatorGroup      bool
+	LogLevel               glog.Level
+}
+
+// OperatorInstaller provides generic operator installation functionality.
+type OperatorInstaller struct {
+	config   OperatorInstallConfig
+	csvUtils *CSVUtils
+}
+
+// NewOperatorInstaller creates a new operator installer with the given configuration.
+func NewOperatorInstaller(config OperatorInstallConfig) *OperatorInstaller {
+	if config.LogLevel == 0 {
+		config.LogLevel = glog.Level(DefaultLogLevel)
+	}
+
+	csvUtils := NewCSVUtils(config.APIClient, config.Namespace, config.LogLevel)
+
+	return &OperatorInstaller{
+		config:   config,
+		csvUtils: csvUtils,
+	}
+}
+
+// Install deploys the operator following the standard OLM pattern.
+func (o *OperatorInstaller) Install() error {
+	glog.V(o.config.LogLevel).Infof("Starting operator installation: %s in namespace %s",
+		o.config.PackageName, o.config.Namespace)
+
+	if err := o.validateConfig(); err != nil {
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	if !o.config.SkipNamespaceCreation {
+		if err := o.createNamespaceIfNotExist(); err != nil {
+			return fmt.Errorf("failed to create namespace: %w", err)
+		}
+
+		glog.V(o.config.LogLevel).
+			Infof("SUCCESS: Namespace %s created/verified",
+				o.config.Namespace)
+	} else {
+		glog.V(o.config.LogLevel).
+			Infof("Skipping namespace creation for global namespace: %s",
+				o.config.Namespace)
+	}
+
+	if !o.config.SkipOperatorGroup {
+		if err := o.createOperatorGroup(); err != nil {
+			return fmt.Errorf("failed to create operator group: %w", err)
+		}
+
+		glog.V(o.config.LogLevel).Infof("SUCCESS: OperatorGroup %s created", o.config.OperatorGroupName)
+	} else {
+		glog.V(o.config.LogLevel).Infof("Skipping operator group creation, using existing: %s", o.config.OperatorGroupName)
+	}
+
+	if err := o.createSubscription(); err != nil {
+		return fmt.Errorf("failed to create subscription: %w", err)
+	}
+
+	glog.V(o.config.LogLevel).Infof("SUCCESS: Subscription %s created", o.config.SubscriptionName)
+	glog.V(o.config.LogLevel).Infof("Operator %s installation completed", o.config.PackageName)
```

Comment:
IIUC, you can't tell if the installation is complete at this point.
You can only see if the subscription is created (and you already reported it)